### DISCUSSION TITLE
MCMC sampling for a Bayesian logistic regression model

### DISCUSSION THREAD

**Discussion by darrenjw** [google-research/dex-lang#977] [Show and tell]
I've been comparing MCMC algorithms for a Bayesian logistic regression model, using different algorithms, languages, and libraries.
https://github.com/darrenjw/logreg
I've now added Dex into the mix, so it might be of interest to some Dex folk. I've currently got RWMH, MALA (with diagonal pre-conditioner) and HMC (with diagonal mass matrix), with and without auto-differentiated gradients. I think it's quite interesting to compare implementations (and performance) across a bunch of different languages.
I've also written some utility functions, including some simple functions for reading and writing TSV files. Note that these examples rely on a Dex build from *today*, including the `parseString` function that Adam kindly added to the prelude (and the augmented library search path).

**Comment by apaszke** [comment]
This is amazing, thanks for sharing! Those programs will be very useful for us as benchmarks. We've recently lost almost all capability to optimize Dex programs, because we've rewritten the compiler from scratch, so this might be a very good driver for progress!

Given that you've now spent some time with Dex, I'd be very curious to hear your thoughts. Are there any things you particularly liked or disliked? We still have lots of flexibility in language design and we keep trying to explore improvements, so any feedback is very welcome and extremely useful for us. I'm also happy to follow up by email if you'd prefer to discuss in a less public channel.

**Reply by apaszke** [reply]
I noticed that you had to hardcode the Dex directory in your `Makefile` to compute the default library search path. This seems quite inconvenient, so I sent https://github.com/google-research/dex-lang/pull/984 which should let you get rid of that.
